src/test/java/com/coveo/spillway/SpillwayTest.java

```diff
@@ -384,4 +394,74 @@ public void triggersAreIgnoreIfTheStorageReturnsAnIncoherentResponse() {
 
     verify(callback, never()).trigger(any(LimitDefinition.class), any());
   }
+
+  @Test
+  public void canAddCapacityLimitOverride() {
+    LimitOverride override = LimitOverrideBuilder.of(JOHN).to(A_CAPACITY).per(A_DURATION).build();
```

Comment:
I talked to @francistb and I like the idea of a Predicate. Still I'm not sure how we should handle duplicate elements since multiple predicates or extractors could be applied for each query...